### ISSUE TITLE
ruby wrapper for imagemagick?

### ISSUE THREAD

**Issue by evanwill** [CollectionBuilder/collectionbuilder-sa_draft#26]
would using a ruby wrapper for ImageMagick, like [minimagick](https://github.com/minimagick/minimagick) help deal with compatibility issues of imagemagick 6 vs 7?

it looks like you can use `convert` in minimagick on both 6 and 7, which is the only command we are really using.

**Comment by derekenos** [comment]
I like the fact that this library would offer a more user-friendly, Ruby interface to ImageMagick, but question whether it's worth adding another dependency.

The [ImageMagick 6 > 7 porting guide](https://imagemagick.org/script/porting.php#cli) makes it look like the only change that impacts our usage is the command name change from `convert` to `magick`.

One way to deal with this would be to have the rake task check the ImageMagick version and invoke `convert` or `magick` as appropriate:
```
require "open3"

output = Open3.capture2e("magick --version")[0]
output.force_encoding(Encoding::UTF_8).match(/^Version: ImageMagick (\d\.\d)/)[1]
```
```
 => "7.0" 
```
Had to add the `force_encoding()` to make it work :roll_eyes: 

Or, simply check for which executable exists:

```
require "mkmf"

find_executable "convert"
# => nil 

find_executable "magick"
 # => "/usr/local/bin/magick" 
```

**Comment by evanwill** [comment]
yeah, as you say, for the command we are using the only difference is using the old `convert` vs `magick`, minimagick is probably doing something like you propose. 
and yeah, probably not worth an extra gem. 

I might be an outlier, but just coming up because most linux repos they keep it as v6. If it is possible to switch out the command in a reasonable way, might be handy to be able to swap in graphicsmagick as well (command `gm convert`). When I created thumbs for big PDF collections, gm was much faster.

**Comment by derekenos** [comment]
I installed `graphicsmagick v1.3.28` and swapped out the `magick` command with `gm` but got this error indicating that the `magick` / `gm` commands are not interchangeable:
> gm: Unrecognized command 'density'. Use -help for a usage summary or see manual..

If you want to create a separate issue for adding `graphicsmagick` support, I will get it done. Hopefully it'll require little more than command line argument juggling but it's an unknown.

I merged in support for specifying a different `ImageMagick` executable (e.g. `convert` for `v6`) here: https://github.com/CollectionBuilder/collectionbuilder-sa_draft/pull/30

Ok to close this @evanwill ?
